### DISCUSSION TITLE
How to disable default modules based on condition

### DISCUSSION THREAD

**Discussion by llagerlof** [starship/starship#4331] [Q&A] [answered]
I want to enable the [php] module, but only when the current directory contains the `composer.json` file.

I tried the config below, but doesn't work. Probably because `disabled` doesn't accept commands like `when`:

```
[php]
disabled = "! test -f composer.json"
```

I tried to set the `when`, but didn't worked:

```
[php]
when = "test -f composer.json"
```

So, how to do it?

**Comment by davidkna** [comment, marked as answer]
It's not possible to add arbitrary conditions without external help, but in this case you can add `!compose.json` to `php.detect_files` if you are using a recent version of starship.

**Reply by llagerlof** [reply]
Thank you very much. 

I added your config suggestion. I copied the file list from the default values in documentation and added a wildcard:

```
[php]
detect_files = ["composer.json", ".php-version", "*php"]
```
And it worked.

[edit: I overwrite my previous reply because I could not reproduce the problem anymore, so i'ts gone for good]

**Reply by davidkna** [reply]
Sorry, I thought you only wanted to do the reverse and never see the mode if there is a `composer.json` file. To get rid of the triggers besides `composer.json`:
```toml
[php]
detect_files = ["composer.json"]
detect_extensions = []
```
